python_modules/dagster/dagster/_grpc/server.py

```diff
@@ -1524,55 +1524,101 @@ def __init__(
+                fixed_server_id=self._fixed_server_id,
+                startup_timeout=self._startup_timeout,
+                cwd=self._cwd,
+                log_level=self._log_level,
+                env=self._env,
+                inject_env_vars_from_instance=self._inject_env_vars_from_instance,
+                container_image=self._container_image,
+                container_context=self._container_context,
+                additional_timeout_msg=self._additional_timeout_msg,
             )
         else:
             self.socket = safe_tempfile_path_unmanaged()
 
             server_process = open_server_process(
-                instance_ref=instance_ref,
-                location_name=location_name,
-                server_command=server_command,
+                instance_ref=self._instance_ref,
+                location_name=self._location_name,
+                server_command=self._server_command,
                 port=None,
                 socket=self.socket,
-                loadable_target_origin=loadable_target_origin,
-                max_workers=max_workers,
-                heartbeat=heartbeat,
-                heartbeat_timeout=heartbeat_timeout,
-                fixed_server_id=fixed_server_id,
-                startup_timeout=startup_timeout,
-                cwd=cwd,
-                log_level=log_level,
-                env=env,
-                inject_env_vars_from_instance=inject_env_vars_from_instance,
-                container_image=container_image,
-                container_context=container_context,
-                additional_timeout_msg=additional_timeout_msg,
+                loadable_target_origin=self._loadable_target_origin,
+                max_workers=self._max_workers,
+                heartbeat=self._heartbeat,
+                heartbeat_timeout=self._heartbeat_timeout,
+                fixed_server_id=self._fixed_server_id,
+                startup_timeout=self._startup_timeout,
+                cwd=self._cwd,
+                log_level=self._log_level,
+                env=self._env,
+                inject_env_vars_from_instance=self._inject_env_vars_from_instance,
+                container_image=self._container_image,
+                container_context=self._container_context,
+                additional_timeout_msg=self._additional_timeout_msg,
             )
 
         if server_process is None:
             raise CouldNotStartServerProcess(port=self.port, socket=self.socket)
         else:
             self._server_process = server_process
 
-        self._wait_on_exit = wait_on_exit
+    def auto_restart_thread(self):
+        while True:
+            time.sleep(1)
+            if self.__shutdown_event.is_set():
+                break
+            if self._server_process and self._server_process.poll() is not None:
+                self.restart_server()
+
+    def restart_server(self) -> None:
+        if self.server_process.poll() is None:
```

Comment:
i think this is the opposite of what you want. poll() (confusingly) returns None whenever the process is running, so this will restart the server every time

Reply:
ah actually this little if statement doesn't make any sense. I was trying to make sure that the underlying grpc server shuts down but this doesn't even do that.

Reply:
oh sorry, I misread

Reply:
But yeah this can't happen because of the code that gates restart_server